File: Backend/Data/ViewModels/Request/CreateUserVM.cs
namespace Backend.Data.ViewModels
{
    /// <summary>
    /// View Model class for Creat User reqeust
    /// </summary>
    public class CreateUserVM
    {
        public string UserName { get; set; }
        public string Email { get; set; }
        public string Name { get; set; }
        public DateTime BirthDate { get; set; }
        public string Password { get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="userName">User's unique identifier</param>
        /// <param name="email">User's email</param>
        /// <param name="name">User's name</param>
        /// <param name="birthDate">User's birth date</param>
        /// <param name="password">User's password</param>
        public CreateUserVM(string userName, string email, string name, DateTime birthDate, string password)
        {
            this.UserName = userName;
            this.Email = email;
            this.Name = name;
            this.BirthDate = birthDate;
            this.Password = password;
        }
    }
}

File: Frontend/src/app/viewModels/requests/CreateUserVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model class for a Create User request
 */
export class CreateUserVM implements toJSONString {

    /**
     * Construct's an instance of {@link CreateUserVM}
     * 
     * @param userName user's user name
     * @param email user's email
     * @param name user's name
     * @param birthDate user's birth date
     * @param password user's password
     */
    constructor(
        public userName: string,
        public email: string,
        public name: string,
        public birthDate: Date,
        public password: string
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    toJSONString(): string {
        return JSON.stringify(this)
    }

}